common/beerocks/bwl/nl80211/ap_wlan_hal_nl80211.cpp

```diff
@@ -0,0 +1,818 @@
+    uint8_t authetication_snr_threshold_hi, uint8_t authetication_snr_threshold_lo)
+{
+    LOG(TRACE) << __func__ << " - NOT IMPLEMENTED!";
+    return false;
+}
+
+bool ap_wlan_hal_nl80211::sta_softblock_remove(const std::string &vap_name,
+                                               const std::string &client_mac)
+{
+    return false;
+}
+
+bool ap_wlan_hal_nl80211::switch_channel(int chan, int bw, int vht_center_frequency)
+{
+    LOG(TRACE) << __func__ << " channel: " << chan << ", bw: " << bw
+               << ", vht_center_frequency: " << vht_center_frequency;
+
+    // TODO: IMPLEMENT!
+
+    LOG(DEBUG) << "Got channel switch, simulate ACS-STARTED;ACS-COMPLETED";
+    event_queue_push(Event::ACS_Started);
+    event_queue_push(Event::ACS_Completed);
+    event_queue_push(Event::CSA_Finished);
+
+    return true;
+}
+
+bool ap_wlan_hal_nl80211::set_antenna_mode(AntMode mode)
+{
+    LOG(TRACE) << __func__ << " - NOT IMPLEMENTED!";
+    return true;
+}
+
+bool ap_wlan_hal_nl80211::wds_set_mode(WDSMode mode)
+{
+    LOG(TRACE) << __func__ << " - NOT IMPLEMENTED!";
+    return true;
+}
+
+bool ap_wlan_hal_nl80211::wds_add_sta(const std::string &mac)
+{
+    LOG(TRACE) << __func__ << " - NOT IMPLEMENTED!";
+    return true;
+}
+
+bool ap_wlan_hal_nl80211::wds_del_sta(const std::string &mac)
+{
+    LOG(TRACE) << __func__ << " - NOT IMPLEMENTED!";
+    return true;
+}
+
+bool ap_wlan_hal_nl80211::wds_clear_list()
+{
+    LOG(TRACE) << __func__ << " - NOT IMPLEMENTED!";
+    return true;
+}
+
+bool ap_wlan_hal_nl80211::failsafe_channel_set(int chan, int bw, int vht_center_frequency)
+{
+    LOG(TRACE) << __func__ << " - NOT IMPLEMENTED!";
+    return true;
+}
+
+bool ap_wlan_hal_nl80211::failsafe_channel_get(int &chan, int &bw)
+{
+    LOG(TRACE) << __func__;
+
+    // Build command string
+    std::string cmd = "GET_FAILSAFE_CHAN";
+    char *reply;
+
+    // Send command
+    if (!wpa_ctrl_send_msg(cmd, &reply)) {
+        LOG(ERROR) << "failsafe_channel_get() failed!";
+        return false;
+    }
+
+    // Custom reply parsing
+    if (!strncmp(reply, "UNSPECIFIED", 11)) {
+        chan = -1;
+        bw   = -1;
+    } else if (!strncmp(reply, "ACS", 3)) {
+        chan = bw = 0;
+
+    } else {
+        int freq;
+        sscanf(reply, "%d %*s %*s bandwidth=%d", &freq, &bw);
+        chan = beerocks::utils::wifi_freq_to_channel(freq);
+    }
+
+    return true;
+}
+
+bool ap_wlan_hal_nl80211::restricted_channels_set(char *channel_list)
+{
+    LOG(TRACE) << __func__ << " - NOT IMPLEMENTED!";
+    return true;
+}
+
+bool ap_wlan_hal_nl80211::restricted_channels_get(char *channel_list)
+{
+    LOG(TRACE) << __func__ << " - NOT IMPLEMENTED!";
+    memset(channel_list, 0, beerocks::message::RESTRICTED_CHANNEL_LENGTH);
+    return true;
+}
+
+bool ap_wlan_hal_nl80211::read_acs_report()
+{
+    LOG(TRACE) << __func__ << " for interface: " << get_radio_info().iface_name;
+
+    // TODO: Get actual list of supported channels
```

Comment:
Please refer to the task which should add it here. Please extend #457 and explain there all that needs to be done for finalizing nl80211 BWL.

Reply:
I'll refer to task #457, but I don't know what to write in the task (what needs to be done for finalizing nl80211 BWL).